lib/bolt/applicator.rb

```diff
@@ -201,7 +196,17 @@ def apply(args, apply_body, scope)
               raise future.reason if future.rejected?
               results = transport.batch_task(batch, catalog_apply_task, arguments, options, &notify)
               Array(results).map do |result|
-                identify_resource_failures(provide_puppet_missing_errors(result))
+                if (puppet_missing = create_puppet_missing_error(result))
```

Comment:
Should this be a class method on the ApplyResult
`ApplyResult.from_task_result(result)`
?

Reply:
That seems reasonable.